ExchangeProxy/UserProxyEngine.cpp

```diff
@@ -13,11 +13,17 @@ UserProxyEngine::UserProxyEngine()
 
 void UserProxyEngine::initializeExchange()
 {
-    int request = listenForRequest();
+    int request = -1;
+    while(request < 0 || request > 6)
+    {
+        printMenu();
+        request = listenForRequest();
+        validateRequest(request);
```

Comment:
validateRequest should return a boolean. If request is valid, then it goes to onSubmitHandler. If it isn't valid, then continue to listen for another request.